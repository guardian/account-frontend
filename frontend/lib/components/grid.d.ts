/// <reference types="react" />
import React from 'react';
export interface ContainerProps {
    readonly children: ReadonlyArray<JSX.Element> | JSX.Element;
}
export declare const Container: React.SFC<ContainerProps>;
