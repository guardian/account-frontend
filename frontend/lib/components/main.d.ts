/// <reference types="react" />
import React from 'react';
export interface MainProps {
    children: JSX.Element[] | JSX.Element;
}
export declare const Main: React.SFC<MainProps>;
