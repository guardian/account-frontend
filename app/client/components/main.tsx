import React from "react";
import palette from "../colours";
import { serif } from "../styles/fonts";
import { Footer } from "./footer/footer";
import Header from "./header";

export interface WithOptionalServerPathWithQueryParams {
  serverPathWithQueryParams?: string;
}

interface MainProps extends WithOptionalServerPathWithQueryParams {
  children: any;
}

export const Main = ({ children }: MainProps) => {
  const tmpIsSignedIn =
    typeof window !== "undefined" && !!window.guardian?.identityDetails?.userId;
  return (
    <div
      css={{
        display: "flex",
        flexDirection: "column",
        height: "100vh",
        alignItems: "stretch",
        width: "100%",
        color: palette.neutral["2"],
      }}
    >
      <Header isSignedOut={!tmpIsSignedIn} />
      <div
        css={{
          flexGrow: 1,
          flexShrink: 0,
          display: "flex",
          flexDirection: "column",
        }}
      >
        <main
          css={{
            fontFamily: serif,
            flexGrow: 1,
            flexShrink: 0,
          }}
        >
          {children}
        </main>
      </div>
      <Footer />
    </div>
  );
};
