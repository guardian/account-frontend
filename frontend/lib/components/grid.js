"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const breakpoints_1 = require("../styles/breakpoints");
const grid_1 = require("../styles/grid");
exports.Container = ({ children }) => react_1.default.createElement("div", { css: {
        position: 'relative',
        margin: 'auto',
        ...grid_1.spanBreakpoints({ mobile: 1, wide: 10 }, breakpoints_1.minWidth),
        ...grid_1.row,
        ['&>*']: grid_1.cell
    } }, children);
