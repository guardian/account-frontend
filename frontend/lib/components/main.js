"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const colours_1 = __importDefault(require("../colours"));
const grid_1 = require("./grid");
exports.Main = ({ children }) => {
    return react_1.default.createElement("div", { css: {
            display: 'flex',
            flexDirection: 'column',
            minHeight: '100vh',
            alignItems: 'stretch',
            width: '100%'
        } },
        react_1.default.createElement("header", { css: {
                backgroundColor: colours_1.default.neutral.header,
                height: '100px',
                color: colours_1.default.neutral["1"],
            } }, "this is the header"),
        react_1.default.createElement("main", { css: { flex: '1' } }, children),
        react_1.default.createElement("footer", { css: {
                backgroundColor: colours_1.default.neutral["1"],
                color: colours_1.default.neutral["7"],
            } },
            react_1.default.createElement(grid_1.Container, null,
                react_1.default.createElement("div", null,
                    react_1.default.createElement("h1", null, "this is the footer")),
                react_1.default.createElement("div", null,
                    react_1.default.createElement("h1", { css: {
                            color: colours_1.default.yellow.medium
                        } }, "this is the fake footer")))));
};
