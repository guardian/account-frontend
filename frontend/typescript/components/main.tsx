import React from 'react';
import { css } from 'emotion'
import { neutral_1, neutral_8, nav_background_colour } from '../colours'
export interface MainProps {
  children: JSX.Element[] | JSX.Element
}

export const Main: React.SFC<MainProps> = ({ children }) => {
  return <div css={{
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
    alignItems: 'stretch',
    width: '100%'
  }}>
    <header css={{
      backgroundColor: nav_background_colour,
      height: '100px',
      color: neutral_1,
    }}>this is the header</header>
    <main css={{ flex: '1' }}>
      {children}
    </main>
    <footer css={{
      backgroundColor: neutral_1,
      color: neutral_8,
    }}>this is the footer</footer>
  </div>
}
