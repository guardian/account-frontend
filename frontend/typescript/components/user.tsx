import { css } from "emotion";
import React from "react";
import { Main } from "./main";

const User = (
  <Main>
    <div css={{ color: "hotpink" }}>Hello world!</div>
  </Main>
);

export default User;

// tslint:disable-next-line:no-expression-statement
// ReactDOM.render(element, document.getElementById('root'));
