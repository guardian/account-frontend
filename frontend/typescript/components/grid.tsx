import { css , cx} from 'emotion'
import React from 'react';
import { minWidth } from '../styles/breakpoints';
import { cell, row, spanBreakpoints } from '../styles/grid';
 export interface ContainerProps {
  readonly children: ReadonlyArray<JSX.Element> | JSX.Element
}

export const Container : React.SFC<ContainerProps> = ({children}) => <div css={{
    position:'relative',
    margin:'auto',
    ...spanBreakpoints({mobile:1,wide:10},minWidth),
    ...row,
    ['&>*']:cell
  }}>
  {children}
  </div>
