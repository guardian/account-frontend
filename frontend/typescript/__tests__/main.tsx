import React from 'react'
import {render, Simulate, wait} from 'react-testing-library'
// this add custom expect matchers from dom-testing-library
import 'dom-testing-library/extend-expect'
import {Main} from '../components/main'

test('Main renders something',()=>{
  let rendered = render(<Main><p>hi</p></Main>)
  expect(rendered.container.firstChild).toMatchSnapshot()
})