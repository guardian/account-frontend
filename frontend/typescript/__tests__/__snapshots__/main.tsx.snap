// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders something 1`] = `
<div
  css="[object Object]"
>
  <header
    css="[object Object]"
  >
    this is the header
  </header>
  <main
    css="[object Object]"
  >
    <p>
      hi
    </p>
  </main>
  <footer
    css="[object Object]"
  >
    this is the footer
  </footer>
</div>
`;
